lms/djangoapps/mfe_api/views.py

```diff
@@ -0,0 +1,52 @@
+"""
+MFE API Views for useful information related to mfes.
+"""
+
+from django.conf import settings
+from django.http import JsonResponse
+from django.utils.decorators import method_decorator
+from django.views.decorators.cache import cache_page
+from rest_framework import status
+from rest_framework.views import APIView
+
+from openedx.core.djangoapps.site_configuration import helpers as configuration_helpers
+
+
+class MFEConfigView(APIView):
+    """
+    Provides an API endpoint to get the MFE_CONFIG from site configuration.
+    """
+
+    @method_decorator(cache_page(settings.MFE_API_CONFIG_CACHE_TIMEOUT))
+    def get(self, request):
+        """
+        GET /api/mfe/v1/config
+        or
+        GET /api/mfe/v1/config?mfe=name_of_mfe
+
+        **GET Response Values**
+        ```
+        {
+            "BASE_URL": "https://name_of_mfe.example.com",
+            "LANGUAGE_PREFERENCE_COOKIE_NAME": "example-language-preference",
+            "CREDENTIALS_BASE_URL": "https://credentials.example.com",
+            "DOSCOVERY_API_BASE_URL": "https://discovery.example.com",
+            "LMS_BASE_URL": "https://courses.example.com",
+            "LOGIN_URL": "https://courses.example.com/login",
+            "LOGOUT_URL": "https://courses.example.com/logout",
+            "STUDIO_BASE_URL": "https://studio.example.com",
+            "LOGO_URL": "https://courses.example.com/logo.png"
+        }
+        ```
+        """
+
+        if not settings.FEATURES.get('ENABLE_MFE_API'):
+            msg = 'MFE API not found. Try setting FEATURES["ENABLE_MFE_API"] to true.'
+            return JsonResponse({'message': msg}, status=status.HTTP_404_NOT_FOUND)
+
+        mfe_config = configuration_helpers.get_value('MFE_CONFIG', {})
+        if request.query_params.get('mfe'):
+            mfe = str(request.query_params.get('mfe')).upper()
+            mfe_config.update(configuration_helpers.get_value(f'MFE_CONFIG_{mfe}', {}))
```

Comment:
We should probably validate that {mfe} is not empty, and that it doesn't contain unsupported characters.

Reply:
Thanks for the feedback. You're right.

Reply:
Hello @arbrandes , I have had little problems when trying to implement this feedback.

- When the mfe is empty it simply assumes that there is no queryparam, then it returns the mfe_config which is when there is no query_param.

- The other thing is that I couldn't find a function or something in the openedx code that would allow me to remove the unsupported characters. I've been researching and I can remove them with a regex but it would be hardcoded like this `regex = re.compile("!$&'()*+,;=")` and I don't think that's very good. I also explored the option of using something like `string.punctuation` instead of using that pattern but I think that would be too restrictive.

I really don't know if the API has to handle those cases, I think the API shouldn't handle the URL encoding.

Any opinion you have on this I would like to know.

Reply:
I would say that with a `.strip` and `.upper` we are sanitizing enough.

I was looking into https://github.com/openedx/edx-platform/search?p=2&q=query_params+get and I dont think the rest of the code does much more.

Reply:
Alright, it works as is.  Thanks for looking into it!